src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -405,6 +405,50 @@ async Task IDurableOrchestrationClient.TerminateAsync(string instanceId, string
             }
         }
 
+        async Task IDurableOrchestrationClient.SuspendAsync(string instanceId, string reason)
+        {
+            OrchestrationState state = await this.GetOrchestrationInstanceStateAsync(instanceId);
+            if (IsOrchestrationSuspendable(state))
+            {
+                state.OrchestrationInstance.ExecutionId = null;
```

Comment:
Making unnecessary state changes like this is not a good practice. It might be safe in this case, but it would be better hygiene to instead create a new OrchestrationInstance object, like this:

```csharp
var instance = new OrchestrationInstance { InstanceId = instanceId };
await this.client.SuspendInstanceAsync(instance, reason);
```

Reply:
Oh, that makes sense to me. Thanks for letting me know!